Libraries/Text/TextInput/RCTBackedTextInputDelegateAdapter.m

```diff
@@ -353,7 +353,7 @@ - (BOOL)textView:(NSTextView *)textView doCommandBySelector:(SEL)commandSelector
   BOOL commandHandled = NO;
   id<RCTBackedTextInputDelegate> textInputDelegate = [_backedTextInputView textInputDelegate];
   // enter/return
-  if (textInputDelegate.textInputShouldReturn && (commandSelector == @selector(insertNewline:) || commandSelector == @selector(insertNewlineIgnoringFieldEditor:))) {
+  if ((commandSelector == @selector(insertNewline:) || commandSelector == @selector(insertNewlineIgnoringFieldEditor:)) && textInputDelegate.textInputShouldReturn) {
```

Comment:
@tom-un, @shergin, @TheSavior: I don't have the full context on why `textInputShouldReturn` also sends a submit event. Is this something we should fix upstream to avoid future surprises?